astropy/units/tests/test_quantity.py

```diff
@@ -1349,3 +1349,16 @@ class MyQuantity(u.Quantity):
     assert type(q1) is u.Quantity
     q2 = u.Quantity(1., my_unit, subok=True)
     assert type(q2) is  MyQuantity
+
+def test_inserting_implicit_Quantity_into_Table():
+    from ... table import Table
```

Comment:
I know I suggested differently before, but looking at this again, I would simplify the test, so that it is clearer what actually went wrong:
```
c = Column(np.arange(10.), unit=u.m)
q = u.Quantity(c)
q[:2] = t[:2]
# AttributeError -> UnitConversionError
```